spec/money_spec.rb

```diff
@@ -388,6 +388,20 @@
     expect(JSON.dump(Money.new(1.00, "CAD"))).to eq('{"value":"1.00","currency":"CAD"}')
   end
 
+  it "creates Money object from JSON Hash in from_json" do
+    one_cad = Money.new(1, "CAD")
+    expect(Money.from_json(one_cad.as_json)).to eq(one_cad)
```

Comment:
It might be worthwhile to have an additional test case where the input is spelled out more explicitly, more as documentation than to guard against breaking changes, e.g.

```ruby
one_cad = Money.new(1, "CAD")
input = '{"amount": 1, "currency": "CAD"}'
expect(Money.from_json(input)).to eq(one_cad)
```